developer/src/kmc-model/src/build-trie.ts

```diff
@@ -185,304 +187,77 @@ function* enumerateLines(lines: string[]): Generator<LineNoAndText> {
-  // by Conrad Irwin.
-  //
-  // trie-ing is distributed under the terms of the MIT license, reproduced here:
-  //
-  //   The MIT License
-  //   Copyright (c) 2015-2017 Conrad Irwin <conrad.irwin@gmail.com>
-  //   Copyright (c) 2011 Marc Campbell <marc.e.campbell@gmail.com>
-  //
-  //   Permission is hereby granted, free of charge, to any person obtaining a copy
-  //   of this software and associated documentation files (the "Software"), to deal
-  //   in the Software without restriction, including without limitation the rights
-  //   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-  //   copies of the Software, and to permit persons to whom the Software is
-  //   furnished to do so, subject to the following conditions:
-  //
-  //   The above copyright notice and this permission notice shall be included in
-  //   all copies or substantial portions of the Software.
-  //
-  //   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-  //   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-  //   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-  //   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-  //   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-  //   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-  //   THE SOFTWARE.
-  //
-  // See: https://github.com/ConradIrwin/trie-ing/blob/df55d7af7068d357829db9e0a7faa8a38add1d1d/LICENSE
-
-  /**
-   * An entry in the prefix trie. The matched word is "content".
-   */
-  interface Entry {
-    content: string;
-    key: SearchKey;
-    weight: number;
-  }
-
-  /**
-   * The trie is made up of nodes. A node can be EITHER an internal node (whose
-   * only children are other nodes) OR a leaf, which actually contains the word
-   * form entries.
-   */
-  type Node = InternalNode | Leaf;
-
-  /**
-   * An internal node.
-   */
-  interface InternalNode {
-    type: 'internal';
-    weight: number;
-    // TODO: As an optimization, "values" can be a single string!
-    values: string[];
-    children: { [codeunit: string]: Node };
-    unsorted?: true;
-  }
-
-  /**
-   * A leaf node.
-   */
-  interface Leaf {
-    type: 'leaf';
-    weight: number;
-    entries: Entry[];
-    unsorted?: true;
-  }
-
-  /**
-   * A sentinel value for when an internal node has contents and requires an
-   * "internal" leaf. That is, this internal node has content. Instead of placing
-   * entries as children in an internal node, a "fake" leaf is created, and its
-   * key is this special internal value.
-   *
-   * The value is a valid Unicode BMP code point, but it is a "non-character".
-   * Unicode will never assign semantics to these characters, as they are
-   * intended to be used internally as sentinel values.
-   */
-  const INTERNAL_VALUE = '\uFDD0';
```

Comment:
This exactly matches the value of `SENTINEL_CODE_UNIT` in `common/models/templates`, so I replaced all migrated instances of it with `SENTINEL_CODE_UNIT`.